### ISSUE TITLE
Cmake error after Upgrade the core?! *My misteak?!*

### ISSUE THREAD

**Issue by Shoxxo** [TrinityCore/TrinityCore#17023]
**Description**:

```quote
Determining if the pthread_create exist failed with the following output:
Change Dir: /home/trinity/TrinityCore/build/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/make" "cmTryCompileExec979401938/fast"
/usr/bin/make -f CMakeFiles/cmTryCompileExec979401938.dir/build.make CMakeFiles/cmTryCompileExec979401938.dir/build
make[1]: Entering directory `/home/trinity/TrinityCore/build/CMakeFiles/CMakeTmp'
/usr/bin/cmake -E cmake_progress_report /home/trinity/TrinityCore/build/CMakeFiles/CMakeTmp/CMakeFiles 1
Building C object CMakeFiles/cmTryCompileExec979401938.dir/CheckSymbolExists.c.o
/usr/bin/cc   -std=gnu99 -W -Wall -Wextra -Winit-self -Winvalid-pch -Wfatal-errors    -o CMakeFiles/cmTryCompileExec979401938.dir/CheckSymbolExists.c.o   -c /home/trinity/TrinityCore/build/CMakeFiles/CMakeTmp/CheckSymbolExists.c
Linking C executable cmTryCompileExec979401938
/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTryCompileExec979401938.dir/link.txt --verbose=1
/usr/bin/cc   -std=gnu99 -W -Wall -Wextra -Winit-self -Winvalid-pch -Wfatal-errors     CMakeFiles/cmTryCompileExec979401938.dir/CheckSymbolExists.c.o  -o cmTryCompileExec979401938 -rdynamic 
CMakeFiles/cmTryCompileExec979401938.dir/CheckSymbolExists.c.o: In function `main':
CheckSymbolExists.c:(.text+0x16): undefined reference to `pthread_create'
collect2: error: ld returned 1 exit status
make[1]: *** [cmTryCompileExec979401938] Error 1
make[1]: Leaving directory `/home/trinity/TrinityCore/build/CMakeFiles/CMakeTmp'
make: *** [cmTryCompileExec979401938/fast] Error 2

File /home/trinity/TrinityCore/build/CMakeFiles/CMakeTmp/CheckSymbolExists.c:
/* */
#include <pthread.h>

int main(int argc, char** argv)
{
  (void)argv;
#ifndef pthread_create
  return ((int*)(&pthread_create))[argc];
#else
  (void)argc;
  return 0;
#endif
}

Determining if the function pthread_create exists in the pthreads failed with the following output:
Change Dir: /home/trinity/TrinityCore/build/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/make" "cmTryCompileExec1750704506/fast"
/usr/bin/make -f CMakeFiles/cmTryCompileExec1750704506.dir/build.make CMakeFiles/cmTryCompileExec1750704506.dir/build
make[1]: Entering directory `/home/trinity/TrinityCore/build/CMakeFiles/CMakeTmp'
/usr/bin/cmake -E cmake_progress_report /home/trinity/TrinityCore/build/CMakeFiles/CMakeTmp/CMakeFiles 1
Building C object CMakeFiles/cmTryCompileExec1750704506.dir/CheckFunctionExists.c.o
/usr/bin/cc   -std=gnu99 -W -Wall -Wextra -Winit-self -Winvalid-pch -Wfatal-errors -DCHECK_FUNCTION_EXISTS=pthread_create   -o CMakeFiles/cmTryCompileExec1750704506.dir/CheckFunctionExists.c.o   -c /usr/share/cmake-3.2/Modules/CheckFunctionExists.c
Linking C executable cmTryCompileExec1750704506
/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTryCompileExec1750704506.dir/link.txt --verbose=1
/usr/bin/cc   -std=gnu99 -W -Wall -Wextra -Winit-self -Winvalid-pch -Wfatal-errors -DCHECK_FUNCTION_EXISTS=pthread_create    CMakeFiles/cmTryCompileExec1750704506.dir/CheckFunctionExists.c.o  -o cmTryCompileExec1750704506 -rdynamic -lpthreads 
/usr/bin/ld: cannot find -lpthreads
collect2: error: ld returned 1 exit status
make[1]: *** [cmTryCompileExec1750704506] Error 1
make[1]: Leaving directory `/home/trinity/TrinityCore/build/CMakeFiles/CMakeTmp'
make: *** [cmTryCompileExec1750704506/fast] Error 2


Performing C++ SOURCE FILE Test boost_filesystem_copy_links_without_NO_SCOPED_ENUM failed with the following output:
Change Dir: /home/trinity/TrinityCore/build/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/make" "cmTryCompileExec1503031870/fast"
/usr/bin/make -f CMakeFiles/cmTryCompileExec1503031870.dir/build.make CMakeFiles/cmTryCompileExec1503031870.dir/build
make[1]: Entering directory `/home/trinity/TrinityCore/build/CMakeFiles/CMakeTmp'
/usr/bin/cmake -E cmake_progress_report /home/trinity/TrinityCore/build/CMakeFiles/CMakeTmp/CMakeFiles 1
Building CXX object CMakeFiles/cmTryCompileExec1503031870.dir/src.cxx.o
/usr/bin/c++    -std=c++11 -W -Wall -Wextra -Winit-self -Winvalid-pch -Wfatal-errors -Woverloaded-virtual -Dboost_filesystem_copy_links_without_NO_SCOPED_ENUM -std=c++11   -o CMakeFiles/cmTryCompileExec1503031870.dir/src.cxx.o -c /home/trinity/TrinityCore/build/CMakeFiles/CMakeTmp/src.cxx
Linking CXX executable cmTryCompileExec1503031870
/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTryCompileExec1503031870.dir/link.txt --verbose=1
/usr/bin/c++    -std=c++11 -W -Wall -Wextra -Winit-self -Winvalid-pch -Wfatal-errors -Woverloaded-virtual -Dboost_filesystem_copy_links_without_NO_SCOPED_ENUM -std=c++11    CMakeFiles/cmTryCompileExec1503031870.dir/src.cxx.o  -o cmTryCompileExec1503031870 -rdynamic -lboost_system -lboost_filesystem -lboost_iostreams 
CMakeFiles/cmTryCompileExec1503031870.dir/src.cxx.o: In function `boost::filesystem::copy_file(boost::filesystem::path const&, boost::filesystem::path const&)':
src.cxx:(.text._ZN5boost10filesystem9copy_fileERKNS0_4pathES3_[_ZN5boost10filesystem9copy_fileERKNS0_4pathES3_]+0x26): undefined reference to `boost::filesystem::detail::copy_file(boost::filesystem::path const&, boost::filesystem::path const&, boost::filesystem::copy_option, boost::system::error_code*)'
collect2: error: ld returned 1 exit status
make[1]: *** [cmTryCompileExec1503031870] Error 1
make[1]: Leaving directory `/home/trinity/TrinityCore/build/CMakeFiles/CMakeTmp'
make: *** [cmTryCompileExec1503031870/fast] Error 2

Source file was:

  #include <boost/filesystem/path.hpp>
  #include <boost/filesystem/operations.hpp>
  int main() { boost::filesystem::copy_file(boost::filesystem::path(), boost::filesystem::path()); }
```
**Current behaviour**: Tell us what happens

Core COmpiler failed by using cmake command! 

**Expected behaviour**: Tell us what should happen instead

```quote
cmake ../ -DCMAKE_INSTALL_PREFIX=/home/trinity/devserver -DCONF_DIR=/home/trinity/devserver/etc -DTOOLS=1 -DWITH_WARNINGS=1
```
**Steps to reproduce the problem**:

1. Use the cmake command
2. Wait a moment
3. Crash

**Branch(es)**: 335/6x  
335 A

**TC hash/commit**:  
9e371b766c75faf899cbbe24515b8b5c01716a50

**TDB version**:  
TDB335.61

**Operating system**:  
Ubuntu 14.04 

**Linking model**: static/dynamic
Whoot ???

[//]: # (This template is for problem reports, for other type of reports edit it accordingly)
[//]: # (If this is a crash report, include the crashlog with https://gist.github.com/)

**Comment by Aokromes** [comment]
Update to Ubuntu 16.04, 14.04 is no longer suported because to old cmake version.

**Comment by Shoxxo** [comment]
Hahaha no, i can not upgrade that is nbot an option, i have many domains databaseses systems etc on the mother server :) Nooo tell me another way

**Comment by wendel841** [comment]
Instead of:

Try:

**Comment by wendel841** [comment]
Instead of:

`apt-get install libboost-dev libboost-thread-dev libboost-system-dev libboost-filesystem-dev libboost-program-options-dev libboost-iostreams-dev`

Try:

`apt-get install libboost1.55-dev libboost-thread1.55-dev libboost-system1.55-dev libboost-filesystem1.55-dev libboost-program-options1.55-dev libboost-iostreams1.55-dev`

That solved problem for me, Ubuntu 14.04

**Comment by ariel-** [comment]
Download new CMake from https://cmake.org/
